### PR TITLE
Auto retry if the spec file is invalid

### LINKED ISSUE #999: Remove broken spec file automatically
If something disrupt the downloading

```
Downloading `crab2.json` to `/root/code/darwinia-network/darwinia/target/x86_64-unknown-linux-gnu/release`
                                         0/1 (0%) eta 00:00:00
━━━━━╾──────────────────────────────────   12.88 MiB/88.59 MiB    114.62 MiB/s eta 1s                                                                      
^C
```

```
Loading genesis from `/root/code/darwinia-network/darwinia/target/x86_64-unknown-linux-gnu/release/crab2.json`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "Error parsing spec file: EOF while parsing a string at line 1 column 14008320"', node/src/chain_spec/mod.rs:158:41
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrac
```

### PR DESCRIPTION
**Pull request by AurevoirXavier**
Closes #999.

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
@boundless-forest please help test locally.

**Comment by boundless-forest** [comment]
```sh
bear@cypress ~/c/darwinia (opt)> ./target/release/darwinia --chain crab --tmp
Downloading `crab2.json` to `/home/bear/coding/darwinia/target/release`
████████████████████████████████████████ 1/1 (100%) eta 00:00:00                                                                                                                                                                          Loading genesis from `/home/bear/coding/darwinia/target/release/crab2.json`
Failed to load genesis from `/home/bear/coding/darwinia/target/release/crab2.json`, starting the `0` retries
Downloading `crab2.json` to `/home/bear/coding/darwinia/target/release`
████████████████████████████████████████ 1/1 (100%) eta 00:00:00                                                                                                                                                                          Loading genesis from `/home/bear/coding/darwinia/target/release/crab2.json`
Failed to load genesis from `/home/bear/coding/darwinia/target/release/crab2.json`, starting the `1` retries
Downloading `crab2.json` to `/home/bear/coding/darwinia/target/release`
████████████████████████████████████████ 1/1 (100%) eta 00:00:00                                                                                                                                                                          Loading genesis from `/home/bear/coding/darwinia/target/release/crab2.json`
Failed to load genesis from `/home/bear/coding/darwinia/target/release/crab2.json`, starting the `2` retries
Downloading `crab2.json` to `/home/bear/coding/darwinia/target/release`
████████████████████████████████████████ 1/1 (100%) eta 00:00:00                                                                                                                                                                          Loading genesis from `/home/bear/coding/darwinia/target/release/crab2.json`
Failed to load genesis from `/home/bear/coding/darwinia/target/release/crab2.json`, starting the `3` retries
Downloading `crab2.json` to `/home/bear/coding/darwinia/target/release`
████████████████████████████████████████ 1/1 (100%) eta 00:00:00                                                                                                                                                                          Loading genesis from `/home/bear/coding/darwinia/target/release/crab2.json`
Failed to load genesis from `/home/bear/coding/darwinia/target/release/crab2.json`, starting the `4` retries
Downloading `crab2.json` to `/home/bear/coding/darwinia/target/release`
████████████████████████████████████████ 1/1 (100%) eta 00:00:00                                                                                                                                                                          Loading genesis from `/home/bear/coding/darwinia/target/release/crab2.json`
Failed to load genesis from `/home/bear/coding/darwinia/target/release/crab2.json`, starting the `5` retries
Downloading `crab2.json` to `/home/bear/coding/darwinia/target/release`
████████████████████████████████████████ 1/1 (100%) eta 00:00:00                                                                                                                                                                          Loading genesis from `/home/bear/coding/darwinia/target/release/crab2.json`
Failed to load genesis from `/home/bear/coding/darwinia/target/release/crab2.json`, starting the `6` retries
thread 'main' panicked at 'Exit after 5 retries', node/src/chain_spec/mod.rs:171:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

File: node/src/chain_spec/mod.rs [hunk 1]
```diff
@@ -155,5 +155,22 @@ where
...
+		c
+	} else {
+		println!("Failed to load genesis from `{f_name}`, starting the `{retries}` retries");
+
+		// Try removing the invalid file.
+		//
+		// Sometimes, it might not exist.
+		let _ = fs::remove_file(f_name);
+
+		if retries > 5 {
```
Comment (boundless-forest) [inline node/src/chain_spec/mod.rs:170]: Start from 0,  0~6, it'll retry 7 times

Comment (AurevoirXavier) [reply node/src/chain_spec/mod.rs]: Can you try it again?

**Review by boundless-forest — APPROVED** [review]

**Pull request merged**